src/transforms/regex_parser.rs

```diff
@@ -208,18 +254,15 @@ impl Transform for RegexParser {
                 }
             };
 
-            let mut capture_locs = match self.capture_locs.get_mut(id) {
-                Some(capture_locs) => capture_locs,
+            let pattern = match self.patterns.get_mut(id) {
+                Some(pattern) => pattern,
                 None => {
                     error!(message = "Cannot find capture locations for pattern", %id, rate_limit_secs = 30);
                     return None;
```

Comment:
This error message looks out of date. Also, we may want to follow up this PR with one that changes the error handling strategy here. It shouldn't be possible to get an `id` from the regex set that doesn't correspond to a pattern.

Reply:
I agree. Would it be appropriate to `panic!` here since it'd indicate a programmer issue that we'd catch early? I'm not sure what vector's idioms for handling this sort of failure are.